FastSurferCNN/run_prediction.py

```diff
@@ -184,56 +184,36 @@ def get_prediction(self):
         if self.view_ops["coronal"]["cfg"] is not None:
             if self.view_ops["axial"]["cfg"] is not None or self.view_ops["sagittal"]["cfg"] is not None:
                 self.set_model("coronal")
-            if not self.small_gpu:
-                pred_prob = torch.zeros((self.dim, self.dim, od, self.model.get_num_classes()),
-                                        dtype=torch.float).to(self.device)
-            else:
-                pred_prob = torch.zeros((self.dim, self.dim, od, self.model.get_num_classes()),
-                                        dtype=torch.float)
+            pred_prob = torch.zeros((self.dim, self.dim, od, self.get_num_classes()),
+                                    dtype=torch.float, device='cpu' if self.small_gpu else self.device)
             pred_prob = self.run_model(pred_prob)
 
         # axial inference
         if self.view_ops["axial"]["cfg"] is not None:
             LOGGER.info("Run axial view agg")
-            if self.view_ops["coronal"]["cfg"] is not None or self.view_ops["sagittal"]["cfg"] is not None:
+            if self.view_ops["coronal"]["cfg"] is not None:
```

Comment:
@dkuegler why has the "or" part been removed here and not in the other views?

Reply:
The condition is that "if you already have allocated the memory" (i.e. pred_prob exists) and it only exists on one of the two "or" clauses.

Reply:
above (l185) and below (l208) it is checking the other two views, here only one. Still don't get why.

Reply:
It should check for both because afterwards the model is set to the appropriate view. If you only run one view, this is not necessary, but as soon as one of the other has been run below, the next subject needs to start at coronal again and should therefore be reset to it (specifically the checkpoints).

Reply:
@dkuegler can you discuss with @LeHenschel and fix this. If I understand correctly it should be "or .." , but this is not yet fixed in the VINN branch.